File: LeetCodeProblemsGo/LeetCode3005CountElementsWithMaximumFrequency/LeetCode3005CountElementsWithMaximumFrequency.go
package LeetCode3005CountElementsWithMaximumFrequency

func maxFrequencyElements(nums []int) int {
	frequencies := make(map[int]int)
	maxFrequency := 0
	maxFreqOccurrences := 0
	for _, num := range nums {
		if _, exists := frequencies[num]; !exists {
			frequencies[num] = 0
		}
		frequencies[num]++
		if frequencies[num] > maxFrequency {
			maxFreqOccurrences = 1
			maxFrequency = frequencies[num]
			continue
		}
		if frequencies[num] == maxFrequency {
			maxFreqOccurrences++
		}
	}
	return maxFreqOccurrences * maxFrequency

}

File: LeetCodeProblemsCpp/LeetCodeProblemsCpp/Problems/LeetCode3005CountElementsWithMaximumFrequency.cpp
#include "LeetCode3005CountElementsWithMaximumFrequency.h"

#include <unordered_map>
using namespace std;

int LeetCode3005CountElementsWithMaximumFrequency::maxFrequencyElements(vector<int>& nums)
{
    unordered_map<int, int> map{};
    int maxFrequency{};
    for (auto num : nums)
    {
        if (!map.contains(num))
        {
            map[num] = 0;
        }
        map[num]++;
        maxFrequency = max(maxFrequency, map[num]);
    }
    int result{};
    for (auto & pair : map)
    {
        if (pair.second != maxFrequency)
            continue;
        result+=pair.second;
    }
    return result;
}
